content/blog/2023/2023-03-31-serialize-deserialize-with-jsonview.md

```diff
@@ -0,0 +1,617 @@
+        mapper.disable(MapperFeature.DEFAULT_VIEW_INCLUSION);
+        return mapper;
+    }
+}
+````
+With this configuration, let us execute the same API call again:
+{{% image alt="settings" src="images/posts/spring-boot-jsonview/API-02.JPG" %}}
+Now, we don't see the `additionalData` field in the JSON response.
+Here, `mapper.disable(MapperFeature.DEFAULT_VIEW_INCLUSION)` applies to both serialization and deserialization process.
+Instead, if we need to **include no view annotation fields only in the serialization process**, we can apply the following configuration:
+````java
+@Configuration
+public class CommonBean {
+
+    @Bean
+    public ObjectMapper objectMapper() {
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false);
+        mapper.registerModule(new JavaTimeModule());
+        //mapper.disable(MapperFeature.DEFAULT_VIEW_INCLUSION);
+        mapper.getSerializationConfig().without(MapperFeature.DEFAULT_VIEW_INCLUSION);
+        //mapper.getDeserializationConfig().without(
+        // MapperFeature.DEFAULT_VIEW_INCLUSION);
+        return mapper;
+    }
+}
+````
+
+## Detailed Working of @JsonView Use Cases
+
+### Allows control over the data exposed
+
+Let us take a look at how `UserSummary` and `UserDetailedSummary` views are used:
+````java
+@RestController
+@RequestMapping("/internal")
+public class InternalUserController {
+
+    private static final Logger log = LoggerFactory.getLogger(
+            InternalUserController.class);
+
+    private final UserService userService;
+
+    public InternalUserController(UserService userService) {
+        this.userService = userService;
+    }
+
+
+    @GetMapping("/userdetails/all")
+    @JsonView(Views.UserDetailedSummary.class)
+    public ResponseEntity<UserData> getDetailUsers(@RequestParam String loginId) {
+        return ResponseEntity.ok().body(userService.getUser(loginId));
+    }
+
+    @GetMapping("/userdetails")
+    @JsonView(Views.UserSummary.class)
+    public ResponseEntity<UserData> getUserSummary(@RequestParam String loginId) {
+        return ResponseEntity.ok().body(userService.getUser(loginId));
+    }
+}
+````
+{{% image alt="settings" src="images/posts/spring-boot-jsonview/userSummary.JPG" %}}
+{{% image alt="settings" src="images/posts/spring-boot-jsonview/userDetailedSummary.JPG" %}}
+
+As seen from the JSON responses, `/internal/userdetails/all` API uses the `UserDetailedSummary` view to return a detailed response in comparison to 
+`/internal/userdetails` that uses the `UserSummary` view having fewer fields.
+
+
+### Separate views for HTTP Request Methods
+
+We can also create a view that caters to PATCH requests, so that only the defined fields get updated in the downstream system.
+In our `UserData` model class, only the below three address fields cater to `PatchView`:
+{{% image alt="settings" src="images/posts/spring-boot-jsonview/patchView.JPG" %}}
```

Comment:
Please add the code as a code block instead of a picture. This would make it more consistent with the formatting of the rest of the document, and readers would be able to copy the code if necessary.

Reply:
Done